File: utils/Optimization/entire_path/generate_polynomial_matrix.m
function matrix = generate_polynomial_matrix(max_exponent,max_diff,var)


matrix = [];

for i = 0:max_diff
    polynomial = [];
    for j = 0:max_exponent
        
        exponent = j - i;
        if exponent >= 0
            term = var^exponent;
        else
            term = 0;
        end
        polynomial = [polynomial term];
        
    end
    matrix = [matrix;polynomial];
end

end

File: rotor_tm_traj/traj/Optimization/entire_path/generate_polynomial_matrix.py
#! /usr/bin/env python

import numpy as np
# test passed


def generate_polynomial_matrix(max_exponent, max_diff, var):
    matrix = np.zeros((0, max_exponent + 1), dtype=float)
    for i in range(0, max_diff + 1):
        polynomial = np.array([[]])
        for j in range(0, max_exponent + 1):
            exponent = j - i
            if exponent >= 0:
                term = var**exponent
            else:
                term = 0
            polynomial = np.append(polynomial, np.array([[term]]), axis=1)
        matrix = np.append(matrix, polynomial, axis=0)
    return matrix
